src/metrics.rs

```diff
@@ -467,7 +469,7 @@ where
     }
 }
 
-impl<S> Layer<S> for MetricsLayer<S>
+impl<S, M: 'static> Layer<S> for MetricsLayer<S, M>
```

Comment:
Why does this need the explicit lifetime?

Reply:
Because `Layer<S>` has a `Self: 'static` clause.